### PR TITLE
Add status quo story about Grace trying to use gdb to step through a program

### PR DESCRIPTION
**Pull request by pnkfelix**
Written at tail end of group writing session hosted by niko, with useful input from other participants. (Especially point that Grace is the only character this really applies to...)

### CODE REVIEW

File: src/vision/grace_waits_for_gdb_next.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+# 😱 Status quo stories: Template
+
+*This is a template for adding new "status quo" stories. To propose a new status quo PR, do the following:*
```
Comment (nikomatsakis) [inline src/vision/grace_waits_for_gdb_next.md:3]: Can you delete the italicized text?

File: src/vision/grace_waits_for_gdb_next.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+Grace is now able to use `next` to walk through the main function. She does notice that the calls to `tokio::spawn` are skipped over by `next`, but that's not as much of a surprise to her, since those are indeed function calls that are taking async blocks. She sets breakpoints on the first line of each async block so that the debugger will stop when control reaches them as she steps through the code.
+
+
+## 🤔 Frequently Asked Questions
+
+*Here are some standard FAQ to get you started. Feel free to add more!*
+
+* **What are the morals of the story?**
+    * A common usage pattern: hitting `next` to go to what seems like the next statement, breaks down due to implementation details of `#[tokio::main]` and `async fn`.
+    * This is one example of where `next` break, in terms of what a user is likely to *want*. The other common scenario where the behavior of `next` is non-ideal is higher-order functions, like `option.and_then(|t| { ... }`, where someone stepping through the code probably *wants* `next` to set
```
Comment (nikomatsakis) [inline src/vision/grace_waits_for_gdb_next.md:101]:
```suggestion
    * This is one example of where `next` breaks, in terms of what a user is likely to *want*. The other common scenario where the behavior of `next` is non-ideal is higher-order functions, like `option.and_then(|t| { ... }`, where someone stepping through the code probably *wants* `next` to set
```

**Review by nikomatsakis — CHANGES_REQUESTED** [review]
Left a nit, but otherwise lgtm.

File: src/vision/grace_waits_for_gdb_next.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+This is a draft "status quo" story submitted as part of the brainstorming period. It is derived from real-life experiences of actual Rust users and is meant to reflect some of the challenges that Async Rust programmers face today. 
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories [cannot be wrong], only inaccurate). Alternatively, you may wish to [add your own status quo story][htvsq]!
+
+## The story
+
+Grace wants to walk through the behavior of a toy program.
+
+She first fires up `cargo run --verbose` to remind himself what the path to the target binary is. Part of the resulting Cargo output is:
```
Comment (rylev) [inline src/vision/grace_waits_for_gdb_next.md:30]:
```suggestion
She first fires up `cargo run --verbose` to remind herself what the path to the target binary is. Part of the resulting Cargo output is:
```

Comment (rylev) [reply src/vision/grace_waits_for_gdb_next.md:30]: Is this level of detail important? Can we just assume Grace knows where the debug binary is? I'm just not sure it adds much to the story.

**Review by rylev — APPROVED** [review]
Short and sweet. Looks good to me 😊

File: src/vision/grace_waits_for_gdb_next.md (hunk 1 above)
Comment (pnkfelix) [reply src/vision/grace_waits_for_gdb_next.md]: done

File: src/vision/grace_waits_for_gdb_next.md (hunk 3 above)
Comment (pnkfelix) [reply src/vision/grace_waits_for_gdb_next.md:11]: Well, when I was doing these steps to demonstrate the problem  during the writing session (and thus literally using `cargo run --verbose` myself, Niko said aloud "ugh why don't we have `cargo gdb`..."

To me, that's a sign that this level of detail *is* relevant.

**Pull request merged**